src/Modal/users/UserModal.js

```diff
@@ -0,0 +1,117 @@
+import React, { Fragment } from "react";
+import Modal from "react-bootstrap/Modal";
+import { FormInput, FormInputAddressField } from "../../common/Input/FormInput";
+import { updateUserRequest, newUserRequest } from "../../services/api";
+import { getUserObject } from "../../utils/helpers/generatorHelper";
+
+const EditUser = ({
+  user,
+  setUser,
+  showModal,
+  setShowModal,
+  users,
+  setUsers
+}) => {
+  let isPopulated = true;
+
+  for (let key in user) {
+    if (user[key] === "") {
+      isPopulated = false;
+      break;
+    }
+  }
+
+  const editOrAdd = async () => {
+    setShowModal(false);
+    if (isPopulated) {
+      const payLoad = user;
+      const usersCopy = [...users];
+      try {
+        const userIndex = usersCopy.findIndex((user) => user.id === payLoad.id);
+        const userNewData = getUserObject(payLoad);
+        usersCopy[userIndex] = userNewData;
+        setUsers([...usersCopy]);
+        await updateUserRequest(payLoad);
+      } catch (error) {
+        setUsers([...users]);
+      }
+    } else {
+      if (user.name) {
```

Comment:
study formik for forms and yup for validations, also these validations needs to be improved